web/server/web_client.c

```diff
@@ -1294,34 +1240,75 @@ static inline int web_client_switch_host(RRDHOST *host, struct web_client *w, ch
+                    host = find_host_by_node_id(tok);
+            }
+        }
+
+        if(!host) {
+            // we didn't find it, but it may be a uuid case mismatch for MACHINE_GUID
+            // so, recreate the machine guid in lower-case.
+            uuid_t uuid;
+            char txt[UUID_STR_LEN];
+            if (uuid_parse(tok, uuid) == 0) {
+                uuid_unparse_lower(uuid, txt);
+                host = rrdhost_find_by_guid(txt);
+            }
+        }
+
+        if (host) {
+            if(!url) { //no delim found
+                debug(D_WEB_CLIENT, "%llu: URL doesn't end with / generating redirect.", w->id);
+                char *protocol, *url_host;
 #ifdef ENABLE_HTTPS
-            protocol = ((w->ssl.conn && !w->ssl.flags) || w->ssl.flags & NETDATA_SSL_PROXY_HTTPS) ? "https" : "http";
+                protocol = ((w->ssl.conn && !w->ssl.flags) || w->ssl.flags & NETDATA_SSL_PROXY_HTTPS) ? "https" : "http";
 #else
-            protocol = "http";
+                protocol = "http";
 #endif
-            url_host = (!w->forwarded_host[0])?w->server_host:w->forwarded_host;
-            buffer_sprintf(w->response.header, "Location: %s://%s%s/\r\n", protocol, url_host, w->last_url);
-            buffer_strcat(w->response.data, "Permanent redirect");
-            return HTTP_RESP_REDIR_PERM;
-        }
 
-        // copy the URL, we need it to serve files
-        w->last_url[0] = '/';
+                url_host = w->forwarded_host;
+                if(!url_host) {
+                    url_host = w->server_host;
+                    if(!url_host) url_host = "";
+                }
 
-        if(url && *url) strncpyz(&w->last_url[1], url, NETDATA_WEB_REQUEST_URL_SIZE - 1);
-        else w->last_url[1] = '\0';
+                buffer_sprintf(w->response.header, "Location: %s://%s/%s/%s/%s",
+                               protocol, url_host, nodeid?"node":"host", tok, buffer_tostring(w->url_path_decoded));
 
-        host = rrdhost_find_by_hostname(tok);
-        if (!host)
-            host = rrdhost_find_by_guid(tok);
-        if (host) return web_client_process_url(host, w, url);
+                if(buffer_strlen(w->url_query_string_decoded)) {
+                    const char *query_string = buffer_tostring(w->url_query_string_decoded);
+                    if(*query_string) {
+                        if(*query_string != '?')
+                            buffer_fast_strcat(w->response.header, "?", 1);
+                        buffer_strcat(w->response.header, query_string);
+                    }
+                }
+                buffer_fast_strcat(w->response.header, "\r\n", 2);
+                buffer_strcat(w->response.data, "Permanent redirect");
+                return HTTP_RESP_REDIR_PERM;
+            }
+
+            buffer_flush(w->url_path_decoded);
+            buffer_fast_strcat(w->url_path_decoded, "/", 1);
+            buffer_strcat(w->url_path_decoded, url);
+            return func(host, w, url);
```

Comment:
This part will overwrite `url`. Because of this the local agent dashboard cannot access children. 

This results in `404` in access.log

eg

```
1591786 '[localhost]:49128' 'DATA' (sent/all = 126/173 bytes -27%, prep/sent/total = 0.09/0.03/0.12 ms) 404 '/host/rasp/api/v1/registry?action=hello'
```